diag_manager/fms_diag_object.F90

```diff
@@ -1131,4 +1131,43 @@ FUNCTION get_diag_obj_from_id ( id ) result (obj_ptr)
     obj_ptr => diag_objs(id)
   END IF
 END FUNCTION get_diag_obj_from_id
+
+!> @brief Add a attribute to the diag_obj using the diag_field_id
+subroutine fms_diag_field_add_attribute(diag_field_id, att_name, att_value)
+  integer,          intent(in) :: diag_field_id      !< Id of the axis to add the attribute to
+  character(len=*), intent(in) :: att_name     !< Name of the attribute
+  class(*),         intent(in) :: att_value(:) !< The attribute value to add
+
+  type(fmsDiagObject_type), pointer :: obj
+
+  obj => get_diag_obj_from_id ( diag_field_id )
+  if (.not. associated(obj)) return
+
+  obj%num_attributes = obj%num_attributes + 1
+  if (obj%num_attributes > max_field_attributes) &
+    call mpp_error(FATAL, "diag_field_add_attribute: Number of attributes exceeds max_field_attributes for field:"&
+                           //trim(obj%varname)//".  Increase diag_manager_nml:max_field_attributes.")
+
+  call obj%attributes(obj%num_attributes)%add(att_name, att_value)
+  nullify(obj)
+end subroutine fms_diag_field_add_attribute
+
+!> @brief Determines the diag_obj id corresponding to a module name and field_name
+!> @return diag_obj id
+INTEGER FUNCTION fms_get_diag_field_id(module_name, field_name)
+  CHARACTER(len=*), INTENT(in) :: module_name !< Module name that registered the variable
+  CHARACTER(len=*), INTENT(in) :: field_name !< Variable name
+
+  integer :: i !< For do loops
+
+  fms_get_diag_field_id = DIAG_FIELD_NOT_FOUND
+  do i = 1, registered_variables
```

Comment:
Would a `where` be better here?  I feel like this is very inefficient but I can't think of anything else that might be better.

Reply:
I don't think there is any way around it. 

Maybe like this. There is no reason to compare the modname if the varname is not the same? 
```F90
if ((diag_objs(i)%get_varname() .eq. trim(field_name)) then
  if ( diag_objs(i)%get_modname() .eq. trim(module_name)) then
    fms_get_diag_field_id = i
    return
  endif
endif
```